core/templates/dev/head/pages/exploration_editor/ExplorationSaveService.js

```diff
@@ -198,10 +198,21 @@ oppia.factory('explorationSaveService', [
       },
 
       discardChanges: function() {
-        var confirmDiscard = confirm(
-          'Are you sure you want to discard your changes?');
-
-        if (confirmDiscard) {
+        $modal.open({
+          templateUrl: 'modals/confirmDiscardChanges',
+          backdrop: 'static',
+          keyboard: false,
+          controller: [
+            '$scope', '$modalInstance', function($scope, $modalInstance) {
+              $scope.cancel = function() {
+                $modalInstance.dismiss();
+              };
+              $scope.discard = function() {
```

Comment:
discardChanges is the name of the function at line 200 , so I have renamed this function as "confirmDiscard" instead.